lib/timezone/zone.rb

```diff
@@ -236,7 +236,7 @@ def binary_search(time, from = 0, to = nil, &block)
 
       return from if from == to
 
-      mid = (from + to) / 2
```

Comment:
Let's use `(from + to).div(2)` instead - that is always the same and performs integer division.

Also, can you include a test that reproduces this bug?